iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber.inl

```diff
@@ -0,0 +1,165 @@
+{
+    if (!m_subscriber.has_value())
+    {
+        return;
+    }
+
+    if (HasReceiveHandler())
+    {
+        UnsetReceiveHandler();
+    }
+
+    m_subscriber.value().unsubscribe();
+    m_subscriber.reset();
+}
+
+template <typename T>
+template <typename Callable>
+inline core::Result<size_t>
+EventSubscriber<T, EventTransmission::IOX>::GetNewSamples(Callable&& callable, size_t maxNumberOfSamples) noexcept
+{
+    core::Result<size_t> numberOfSamples{0};
+
+    if (!m_subscriber.has_value())
+    {
+        return numberOfSamples;
+    }
+
+    // Hint: Depending on the type of callable, it needs to be checked for null or restricting to lambdas
+
+    while (m_subscriber.value()
+               .take()
+               .and_then([&](const auto& sample) {
+                   callable(sample.get());
+                   numberOfSamples++;
+               })
+               .or_else([](auto& result) {
+                   if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                   {
+                       std::cerr << "Error receiving chunk!" << std::endl;
+                   }
+               })
+           && numberOfSamples < maxNumberOfSamples)
+    {
+    }
+    return numberOfSamples;
+}
+
+//! [EventSubscriber setReceiveHandler]
+template <typename T>
+inline void EventSubscriber<T, EventTransmission::IOX>::SetReceiveHandler(EventReceiveHandler handler) noexcept
+{
+    if (!handler)
+    {
+        std::cerr << "Can't attach empty receive handler!" << std::endl;
+        return;
+    }
+    if (!m_subscriber.has_value())
+    {
+        std::cerr << "Call Subscribe() before setting a receive handler!" << std::endl;
+        return;
+    }
+
+    m_listener
+        .attachEvent(m_subscriber.value(),
+                     iox::popo::SubscriberEvent::DATA_RECEIVED,
+                     iox::popo::createNotificationCallback(onSampleReceivedCallback, *this))
+        .expect("Unable to attach subscriber!");
+    m_receiveHandler->emplace(handler);
+}
+//! [EventSubscriber setReceiveHandler]
+
+template <typename T>
+inline void EventSubscriber<T, EventTransmission::IOX>::UnsetReceiveHandler() noexcept
+{
+    if (!m_subscriber.has_value())
+    {
+        return;
+    }
+
+    m_listener.detachEvent(m_subscriber.value(), iox::popo::SubscriberEvent::DATA_RECEIVED);
+    m_receiveHandler->reset();
+}
+
+template <typename T>
+inline bool EventSubscriber<T, EventTransmission::IOX>::HasReceiveHandler() const noexcept
+{
+    auto receiveHandlerGuard = m_receiveHandler.getScopeGuard();
+    return receiveHandlerGuard->has_value() && receiveHandlerGuard->value();
+}
+
+//! [EventSubscriber invoke callback]
+template <typename T>
+inline void EventSubscriber<T, EventTransmission::IOX>::onSampleReceivedCallback(iox::popo::Subscriber<T>*,
+                                                                                 EventSubscriber* self) noexcept
+{
+    if (self == nullptr)
```

Comment:
This should be impossible. The only one attaching callbacks to the listener is `SetReceiveHandler` and it never sets a `nullptr`. So in the end this would be untestable code.

But you can leave it here, I have no strong feeling about it.

Reply:
As developers tend to copy and paste the examples and use it as a starting point for development, I would prefer if we keep the extra sanity check.